docs/USAGE-GUIDE.md

```diff
@@ -1777,6 +1777,216 @@ handle. The handle is considered invalid after **s2n_connection_free** is used.
+
+**s2n_offered_psk_free** frees the memory associated with the **s2n_offered_psk** object.
+
+The following APIs are called by the server to selects the PSK to be used for the connection.
+
+```c
+int s2n_offered_psk_get_identity(struct s2n_offered_psk *psk, uint8_t** identity, uint16_t *size);
+bool s2n_offered_psk_list_has_next(struct s2n_offered_psk_list *psk_list);
+int s2n_offered_psk_list_next(struct s2n_offered_psk_list *psk_list, struct s2n_offered_psk *psk);
+int s2n_offered_psk_list_reread(struct s2n_offered_psk_list *psk_list);
+int s2n_offered_psk_list_choose_psk(struct s2n_offered_psk_list *psk_list, struct s2n_offered_psk *psk);
+typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn, void *context,
+                                          struct s2n_offered_psk_list *psk_list);
+int s2n_config_set_psk_selection_callback(struct s2n_config *config, s2n_psk_selection_callback cb, void *context);
+```
+
+Use the **s2n_config_set_psk_selection_callback** only when a custom PSK selection logic is desired by the client.
+By default, s2n-tls server chooses the first identity in its PSK list that also appears in the client's PSK list.
+
+**s2n_offered_psk_get_identity** gets the PSK identity and PSK identity length for a given offered PSK object.
+Callers can use this API to obtain the identity value set on the **s2n_offered_psk** object.
+
+**s2n_offered_psk_list_has_next** checks whether the offered PSK list has an offered psk object next in line in the list.
+An offered PSK list contains all the pre-shared keys offered by the client for the server to select.
+If an offered PSK exists next in line, this API returns True, otherwise False.
+Use this API prior to calling **s2n_offered_psk_list_next** to obtain the next offered PSK object from the offered PSK list.
+
+**s2n_offered_psk_list_next** obtains the next offered PSK object from the offered PSK list.
+Use **s2n_offered_psk_list_has_next** prior to this API call to determine if an offered PSK exists next in the list.
+
+**s2n_offered_psk_list_reread** returns the offered PSK list to its original read state.
+Calls to **s2n_offered_psk_list_next** will start over from the beginning of the PSK list. After **s2n_offered_psk_list_reread** is called, the next call to **s2n_offered_psk_list_next** will return the first PSK in the offered PSK list.
+
+**s2n_offered_psk_list_choose_psk** selects a PSK from the offered PSK list to be used for the connection.
+This API matches the PSK identity received from the client against the server's known PSK identities list,
+in order to choose the PSK to be used for the connection. If the PSK identity sent from the client is NULL,
+no PSK is chosen for the connection. If the client offered PSK identity has no matching PSK identity with the server,
+an error will be returned. Use this API along with the **s2n_psk_selection_callback** to select a PSK for the connection.
```

Comment:
I though **s2n_psk_selection_callback** usage was optional only if one wanted to override the logic? Maybe re-state it

Reply:
Yes, **s2n_psk_selection_callback** is to be used only when the default option in undesired. This APIs are enables the customer for the use-case when **s2n_psk_selection_callback** is present. I think restating that the setting of callback is optional is not adding any value here. Maybe:

```
Use this API along with the optional **s2n_psk_selection_callback** to select a PSK for the connection.
```